src/components/Buttons/utils/ButtonStyle.utils.test.ts

```diff
@@ -0,0 +1,166 @@
+import * as ButtonStyleUtils from './ButtonStyle.utils';
+import {
+    TextButtonProps,
+    ButtonVariant,
+    DefaultButtonThemes,
+    ButtonShapes,
+} from '../base';
+import { Colors } from '../../../config';
+
+const {
+    getInitialBackgroundColor,
+    getBorderRadius,
+    getInitialBorderColor,
+    getInitialTextColor,
+} = ButtonStyleUtils;
+
+describe('Running tests for ButtonStyle utils', () => {
```

Comment:
I did it this way because I had to pass the module into jest.spyOn() as the first argument. That's why I imported everything from ButtonStyle.utils and gave it an alias that I could use to reference the module.

```js
test('Given variant of outline, when getInitialTextColor is called, getInitialBorderColor is called', () => {
        const props: TextButtonProps = {
            buttonVariant: ButtonVariant.OUTLINE,
        };
        const spy = jest.spyOn(ButtonStyleUtils, 'getInitialBorderColor');

        getInitialTextColor(props);

        expect(spy).toHaveBeenCalledTimes(1);
        expect(spy).toHaveBeenCalledWith(props);

        spy.mockRestore();
    });
```

Reply:
Will it not work with a named import?
(Genuine should just an object, same as what you've got going now)

Reply:
Do you mean something like this?
```js
import * as ButtonStyleUtils from './ButtonStyle.utils';
import {
    getInitialBackgroundColor,
    getBorderRadius,
    getInitialBorderColor,
    getInitialTextColor,
} from './ButtonStyle.utils';
```

Reply:
Yea? I haven't tried it for this purpose. I know this syntax works for imports (named import syntax).
But I'm just not sure if your saying there is some issue with using jest like this?

Reply:
Yeah initially, I just had the named import. However for jest.spyOn(), I needed to pass into it the module whose method it will spy on. That's why I needed some way to reference the entire ButtonStyle.utils module. I hope I'm making sense haha.
![Capture](https://user-images.githubusercontent.com/73725297/157169093-10ba8025-4fb9-456d-b26d-da549439fb1d.PNG)

Reply:
Got it